Initial classes

mecha opened the pull request.

mecha committed:
Updated dependencies

mecha committed:
Added PHPS test config

mecha committed:
Imported traits from old abstract package

mecha committed:
Added `AbstractWpdbResourceModel`

mecha committed:
Fixed test names

mecha committed:
Added `WpdbSelectResourceModel`

mecha committed:
Added `WpdbInsertResourceModel`

mecha committed:
Added `WpdbUpdateResourceModel`

mecha committed:
Added `WpdbDeleteResourceModel`

mecha committed:
Updated dependencies

mecha committed:
Added `SearchExpressionTermsCapableTrait`

mecha committed:
Renamed methods for compatibility in ...

... `NormalizeWpPostDataArrayCapableTrait`

mecha committed:
Added ordering, limits and offsets in WPDB RMs

mecha committed:
Updated dependencies

mecha committed:
Added bulk/single insertion switch ...

The WPDB insertion RM can switch between inserting all records in bulk,
or iterating and inserting them one-by-one.

Resolves #2

mecha committed:
WPDB insertion RM returns inserted IDs

mecha committed:
Fixed WPDB insert RM ...

- Added missing trait
- Added bulk insertion flag

mecha committed:
Updated dependencies

mecha committed:
Updated dependency versions

mecha committed:
WPDB query preparation skipped if no args

Because WordPress triggers a notice if `$wpdb->prepare()` is called with a query without placeholders.
Just WordPress things ...

Also amends the return type of the trait method to align with `$wpdb->query()`.

mecha committed:
Added `GetWpdbQueryResultsCapableTrait`

mecha committed:
Fixed trait order in `AbstractWpdbResourceModel`

mecha committed:
`AbstractWpdbResourceModel` uses `GetWpdbQueryResultsCapableTrait`

mecha committed:
Fixed WPDB query result fetching

XedinUnknown committed:
Now depending on WP core

XedinUnknown committed:
Added Phan config

XedinUnknown committed:
Updated PHPS config

XedinUnknown committed:
Code standards fixed

XedinUnknown commented:
The file [report.txt][] contains errors reported by Phan at this moment, after WP core has been installed (WP type declarations), code standards fixed (unused imports etc.), and Phan set to ignore interface `@throws`.

[report.txt]: https://github.com/RebelCode/wp-cqrs-resource-models/files/1882708/report.txt

mecha committed:
Updated dependency version constraints

mecha committed:
Merge branch 'task/initial-classes' of https://github.com/rebelcode/wp-cqrs-resource-models into task/initial-classes

mecha committed:
Updated dependencies

mecha committed:
WPDB Select resource model can now alias tables

mecha committed:
Fixed WPDB SELECT RM to properly pass column map

mecha committed:
Extacted WPDB CQRS RM code into abstract bases

mecha committed:
WPDB preparation args are no longer repeated

mecha committed:
Hash-to-value maps are generated for WPDB, and ...

... the RMs generate SQL builder tokens and value lists for WPDB.

The hash keys generated for the values ensure that duplicate values get different hashes.
This is achieved by using their position as part of the hash.

The RMs create a value-to-token map from the hash-to-value map.
Since the values are now keys, duplicate values get "merged".
This is ok, since the tokens are only used for replacement by the query builder traits.

Finally, the RMs pass a numeric array containing only the values to WPDB.
This list is generated from the hash-to-value map, which processes values in order.
Therfore, thee order of the values should be preserved.

mecha committed:
Fixed UPDATE RM order of placeholders and values

mecha committed:
Fixed WPDB query preparation

mecha committed:
Can handle null values for hash map

mecha committed:
Added more null value handling

mecha committed:
Updated dependencies

mecha committed:
Added iterable normalization traits

mecha committed:
Fixed abstract class not declared abstract

mecha committed:
Updated dependencies

XedinUnknown committed:
Removed extra composer config

XedinUnknown committed:
Now also building on PHP 7.2

XedinUnknown committed:
Increased dep stability; dependencies updated

XedinUnknown committed:
Added changelog entry

mecha committed:
Added unique prefixes to random values in some tests

mecha committed:
Fided division by zero in test

mecha committed:
Merge branch 'task/initial-classes' of https://github.com/rebelcode/wp-cqrs-resource-models into task/initial-classes

mecha committed:
Fixed test not accounting for non-unique random values
